### PR TITLE
Fix NPE in RemoteTaskRunner when some tasks in ZooKeeper but not in Overlord

### LINKED ISSUE #5506: NPE while processing taskAnnouncement in RemoteTaskRunner
Introduced in #5260.

```
2018-03-20T01:11:54,944 ERROR [Curator-PathChildrenCache-1] io.druid.indexing.overlord.RemoteTaskRunner - Failed to handle new worker status: {class=io.druid.indexing.overlord.RemoteTaskRunner, exceptionType=class java.lang.NullPointerException, exceptionMessage=null, worker=jihoons-mbp:8091, znode=/druid/indexer/status/jihoons-mbp:8091/parallelIndexSinglePhaseSubIndex_lineitem_2018-03-20T01:11:54.927Z}
java.lang.NullPointerException
        at io.druid.indexing.overlord.RemoteTaskRunner$6.childEvent(RemoteTaskRunner.java:971) [druid-indexing-service-0.13.0-SNAPSHOT.jar:0.13.0-SNAPSHOT]
        at org.apache.curator.framework.recipes.cache.PathChildrenCache$5.apply(PathChildrenCache.java:520) [curator-recipes-4.0.0.jar:4.0.0]
        at org.apache.curator.framework.recipes.cache.PathChildrenCache$5.apply(PathChildrenCache.java:514) [curator-recipes-4.0.0.jar:4.0.0]
        at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:93) [curator-framework-4.0.0.jar:4.0.0]
        at org.apache.curator.shaded.com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:296) [curator-client-4.0.0.jar:?]
        at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:85) [curator-framework-4.0.0.jar:4.0.0]
        at org.apache.curator.framework.recipes.cache.PathChildrenCache.callListeners(PathChildrenCache.java:512) [curator-recipes-4.0.0.jar:4.0.0]
        at org.apache.curator.framework.recipes.cache.EventOperation.invoke(EventOperation.java:35) [curator-recipes-4.0.0.jar:4.0.0]
        at org.apache.curator.framework.recipes.cache.PathChildrenCache$9.run(PathChildrenCache.java:771) [curator-recipes-4.0.0.jar:4.0.0]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_161]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_161]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_161]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_161]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_161]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_161]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_161]
```

### PR DESCRIPTION
**Pull request by jihoonson**
Fixes #5506.

---
This change is [Reviewable](https://reviewable.io/reviews/druid-io/druid/5511)

### CODE REVIEW

**Comment by clintropolis** [comment]
:+1: We should call out in release notes that there will be a small per task increase in the amount of data stored in zookeeper.

File: indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java [hunk 1]
```diff
@@ -968,7 +968,7 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event)
                             announcement.getTaskType(),
                             zkWorker.getWorker(),
                             TaskLocation.unknown(),
-                            runningTasks.get(taskId).getDataSource()
+                            announcement.getTaskDataSource()
```
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java]: If the task has already been shutdown, will this make it start back up again?

Comment (drcrallen) [reply indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java]: I guess there's already a "we'll add it if we don't know about it" thing immediately below, so the check must be elsewhere.

**Comment by drcrallen** [comment]
@jihoonson can you record in this PR where that assurance is made... prior shutdown tasks should be terminated instead of added to the active queue. I forget where it is and its not easy to find.

**Comment by jihoonson** [comment]
@drcrallen if you mean `TaskQueue` by the active queue, `TaskQueue` is a waiting queue for tasks. When a task becomes ready, it calls `TaskRunner.run()` to run a task. Once a task is added to `TaskRunner`, it doesn't added back to `TaskQueue`. 

When the `remoteTaskRunner` sees an unknown prior shutdown task, it adds the task to its `runningTasks` (https://github.com/druid-io/druid/blob/master/indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java#L966-L983) and immediately completes that task (https://github.com/druid-io/druid/blob/master/indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java#L994-L996).

**Comment by jihoonson** [comment]
> there will be a small per task increase in the amount of data stored in zookeeper.

@clintropolis thanks for the review. This also sounds good.

**Comment by jon-wei** [comment]
@drcrallen did you have any more comments on this PR?

**Pull request merged**